app/uk/gov/hmrc/helptosave/models/PayePersonalDetails.scala

```diff
@@ -104,15 +105,15 @@ object PayePersonalDetails {
         JsError("No Address found in the DES response")
       )(v ⇒
           for {
-            line1 ← (v \ "line1").validate[String]
-            line2 ← (v \ "line2").validate[String]
-            line3 ← (v \ "line3").validateOpt[String]
-            line4 ← (v \ "line4").validateOpt[String]
-            line5 ← (v \ "line5").validateOpt[String]
-            postcode ← (v \ "postcode").validate[String]
-            countryCode ← (v \ "countryCode").validateOpt[Int]
+            line1 ← lookup("line1", v).validate[String]
```

Comment:
You ought to call this method in the lookups for Name and DOB, otherwise an error in either of those would cause the same problem, I think

Reply:
Had a look at Name and DOB, don't leak PII if there is any missing data, as implemented in a different way, so current implementation can be as is.